omniledger/service/service.go

```diff
@@ -421,13 +417,69 @@ func (s *Service) db() *skipchain.SkipBlockDB {
 	return s.skService().GetDB()
 }
 
+func (s *Service) loadConfig(scID skipchain.SkipBlockID) (*Config, error) {
+	coll := s.getCollection(scID)
+	// Find the genesis-darc ID.
+	val, contract, err := coll.GetValueContract(GenesisReferenceID.Slice())
+	if err != nil {
+		return nil, err
+	}
+	if string(contract) != ContractConfigID {
+		return nil, errors.New("did not get " + ContractConfigID)
+	}
+	if len(val) != 64 {
+		return nil, errors.New("value has a invalid length")
+	}
+	// Use the genesis-darc ID to create the config key and read the config.
+	configID := ObjectID{
+		DarcID:     darc.ID(val[:32]),
+		InstanceID: OneNonce,
+	}
+	val, contract, err = coll.GetValueContract(configID.Slice())
+	if err != nil {
+		return nil, err
+	}
+	if string(contract) != ContractConfigID {
+		return nil, errors.New("did not get " + ContractConfigID)
+	}
+	config := Config{}
+	err = protobuf.Decode(val, &config)
+	if err != nil {
+		return nil, err
+	}
+	return &config, nil
+}
+
+func (s *Service) loadBlockInterval(scID skipchain.SkipBlockID) (time.Duration, error) {
+	config, err := s.loadConfig(scID)
+	if err != nil {
+		return defaultInterval, err
+	}
+	return config.BlockInterval, nil
+}
+
+func (s *Service) loadLatestDarc(sid skipchain.SkipBlockID, dID darc.ID) (*darc.Darc, error) {
+	colldb := s.getCollection(sid)
+	if colldb == nil {
+		return nil, fmt.Errorf("collection for skipchain ID %s does not exist", sid.Short())
+	}
+	value, contract, err := colldb.GetValueContract(padKey(dID))
```

Comment:
We need to find a better name for `padKey` - I wrote it when testing. Shall we call it `getDarcKey`?